### ISSUE TITLE
System.IO.FileNotFoundException: 'Could not load file or assembly 

### ISSUE THREAD

**Issue by yuessir** [0xd4d/dnSpy#1104]
Before clicking `Submit new issue`, try the [latest build](https://ci.appveyor.com/project/0xd4d/dnspy/branch/master/artifacts)

- What version did you use?
dnSpy (6.0.4.0)
- Write down all steps needed to reproduce this issue. Also attach any needed files.
1. Load an original DLL file to dnSpy 
![issue_1](https://user-images.githubusercontent.com/8326310/55138651-f2024c00-516e-11e9-835a-e4dff52ac9fe.png)
2. Edit method C# likes changing the DateTime(2019,1,1) to DateTime(2099,1,1).
3. Saving the module.
4. Refer the patched module(DLL) to my console project( .Net core project)
5. Debug and throw the error likes:
```
System.IO.FileNotFoundException
  HResult=0x80070002
  Message=Could not load file or assembly 'RFCworker, Version=1.0.0.2, Culture=neutral, PublicKeyToken=ab103

05c2d0c44gf'. 。
  Source=<Cannot evaluate the exception source>
  StackTrace:
<Cannot evaluate the exception stack trace>
```
Did I miss something?Thanks any suggestions.

**Comment by 0xd4d** [comment]
Make sure RFCworker is in the same dir as the dll file. Also verify that it's in the .deps.json file.

**Comment by yuessir** [comment]
@0xd4d 
I am sure RFCworker is in the same dir as the dll file. And the .deps.json file likes:
```
{
  "runtimeTarget": {
    "name": ".NETCoreApp,Version=v2.2",
    "signature": "da39a3ee5e6b4b0d3255bfef95601890afd80709"
  },
  "compilationOptions": {},
  "targets": {
    ".NETCoreApp,Version=v2.2": {
      "TEST_AP_NCO_Core/1.0.0": {
        "dependencies": {
          "RFCworker": "1.0.0.2"
        },
        "runtime": {
          "TEST_AP_NCO_Core.dll": {}
        }
      },
      "RFCworker/1.0.0.2": {
        "runtime": {
          "RFCworker.dll": {
            "assemblyVersion": "1.0.0.2",
            "fileVersion": "1.0.0.2"
          }
        }
      }
    }
  },
  "libraries": {
    "TEST_AP_NCO_Core/1.0.0": {
      "type": "project",
      "serviceable": false,
      "sha512": ""
    },
    "RFCworker/1.0.0.2": {
      "type": "reference",
      "serviceable": false,
      "sha512": ""
    }
  }
}
```
and the .proje file likes:
  <ItemGroup>
    <Reference Include="RFCworker">
      <HintPath>..\dll\RFCworker.dll</HintPath>
    </Reference>
  </ItemGroup>
so I don't know what happened in the simple case. :{

**Comment by 0xd4d** [comment]
Add complete a repro, it's not possible to repro this with the steps above.

**Comment by yuessir** [comment]
Thanks, I found the problem.
The original DLL file name  is "xxx".
After patched it ,I  saved the module and rename it to "xxx_patched".
I  referenced the patched DLL(xxx_patched.dll) to my project, the build works.
 I got the error when I debug it.
https://social.msdn.microsoft.com/Forums/vstudio/en-US/2ac83a6c-f3b6-4ae6-ae8b-2f7b6916ae41/unable-to-load-file-or-assembly-after-renaming-and-referencing-the-dll-build-works-fine?forum=csharpgeneral
